### PR TITLE
Added a 'one-click' deploy-validator.sh script

### PR DESCRIPTION
**Pull request by christos-h**
## Motivation

Validator on-boarding could get a bit tedious. We received feedback for a script that would take care of the entire deployment process.

## Proposal

Add a script, `deploy-validator.sh` which validators can optionally use to deploy a validator in preparation to joining the network.

### CODE REVIEW

File: scripts/deploy-validator.sh [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+# Function to check if a command exists
+command_exists() {
+    command -v "$1" >/dev/null 2>&1
+}
+
+docker_compose_plugin_installed() {
+    docker compose version >/dev/null 2>&1
+}
+
+if ! command_exists cargo; then
```
Comment (ma2bd) [inline scripts/deploy-validator.sh:14]:
```
if ! [ -x /path/to/file ]; then
  echo "File is an existing executable."
fi
```

File: scripts/deploy-validator.sh [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+if [ -z "$1" ]; then
+  echo "Usage: $0 <host>"
+  exit 1
+fi
+
+HOST="$1"
+
+# Get the current branch name and replace underscores with dashes
+BRANCH_NAME=$(git rev-parse --abbrev-ref HEAD)
+FORMATTED_BRANCH_NAME="${BRANCH_NAME//_/-}"
```
Comment (ma2bd) [inline scripts/deploy-validator.sh:44]: Do we need to compute this?

File: scripts/deploy-validator.sh [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+# Download genesis configuration
+echo "Downloading genesis configuration..."
+wget -O $GENESIS_CONFIG $GENESIS_URL
+
+cd docker
+
+# Generate validator keys
+echo "Generating validator keys..."
+PUBLIC_KEY=$(linera-server generate --validators validator-config.toml)
```
Comment (ma2bd) [inline scripts/deploy-validator.sh:91]: Where is the tool `linera-server` coming from?

**Review by ma2bd — APPROVED** [review]
Maybe we should explain that this script is an example based on the manual? (It's not tested in CI and may break in the future.)

File: scripts/deploy-validator.sh (hunk 3 above)
Comment (christos-h) [reply scripts/deploy-validator.sh:91]: It's part of the `cargo install -p linera-service` binaries right?

**Pull request merged**